cmds/core/netcat/netcat.go

```diff
@@ -386,8 +404,8 @@ func run(args []string) error {
 	// connection mode options
 	fs.BoolVar(&f.zeroIo, "z", false, "zero-I/O mode, report connection status only")
 
-	fs.StringVar(&f.sourcePort, "source-port", netcat.DEFAULT_SOURCE_PORT, "Specify source port to use")
-	fs.StringVar(&f.sourcePort, "p", netcat.DEFAULT_SOURCE_PORT, "Specify source port to use (shorthand)")
+	fs.StringVar(&f.sourcePort, "source-port", "", "Specify source port to use")
```

Comment:
why not have this default to netcat.DEFAULT_SOURCE_PORT? Then you would not need that if on line 90 above?
Do we really need all these (shorthand) in the flags? It's kind of obvious it is the shorthand, as it is one letter?

Reply:
The motivation was because of this check:
```
			if f.sourcePort != "" {
				return nil, fmt.Errorf("%w: got more than one port specification", os.ErrInvalid)
			}
```
If in listen mode both host and port are given as arguments, we would not like to receive an additional source port argument via the flag. But if we let the value have a default, this flag is always set to something not equal to "" and we cannot recognize if any value is passed.

Reply:
If we switch to a default we can only recognize if something else than the default was passed.